contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,661 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.17;
+
+// Created By: Art Blocks Inc.
+
+import "./interfaces/0.8.x/IAdminACLV0.sol";
+import "./interfaces/0.8.x/IGenArtDependencyConsumer.sol";
+
+import "@openzeppelin-4.7/contracts/utils/Strings.sol";
+import "@openzeppelin-4.7/contracts/access/Ownable.sol";
+import "@openzeppelin-4.7/contracts/utils/structs/EnumerableSet.sol";
+import "./libs/0.8.x/BytecodeStorage.sol";
+import "./libs/0.8.x/Bytes32Strings.sol";
+
+/**
+ * @title Art Blocks Dependency Registry, V0.
+ * @author Art Blocks Inc.
+ * @notice Privileged Roles and Ownership:
+ * Permissions managed by ACL contract
+ */
+contract DependencyRegistryV0 is Ownable {
+    using BytecodeStorage for string;
+    using BytecodeStorage for address;
+    using Bytes32Strings for bytes32;
+    using Strings for uint256;
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    uint8 constant AT_CHARACTER_CODE = uint8(bytes1("@")); // 0x40
+
+    /// admin ACL contract
+    IAdminACLV0 public adminACLContract;
+
+    event ProjectDependencyOverrideAdded(
+        address indexed _coreContractAddress,
+        uint256 indexed _projectId,
+        bytes32 _dependencyTypeId
```

Comment:
@jakerockland @ryley-o Do you think it makes sense to always emit the dependencyType as a string instead of bytes?

Reply:
@yoshiwarab I would say so ser–I think that would be a bit more idiomatic, WDYT @ryley-o ?

Reply:
Okay, going to leave as is for simplicity plus the ability to set _dependencyTypeId as an indexed event parameter.